### DISCUSSION TITLE
Managing goals

### DISCUSSION THREAD

**Discussion by kwoot** [GothenburgBitFactory/taskwarrior#2440] [Ideas]
I guess the chances this gets implemented are slim to none, but nonetheless I would like to share with you a path of growth for TW.

It would help if you would take a short look at https://angstrom.life. it is a sort of PM tool to not manage a work project but your life.

In short:
-You define a few life goals. Things like: I want to retire at age 20, I want to loose weight, I want to live more relaxed, I want a steady income, etc.
-Every goal has one or more milestones.
-Every milestone contains one or more projects (similar to TW projects).
-And every project contains one or more tasks (similar to TW tasks).  
-Using time registration (pretty crude in Angstrom) the program will occasionally analyze your progress towards your goals and give you advice.

In use-case lingo: as a TW user I like to connect projects to milestones and milestones to task. When I combine this with TimeWarrior data I can analyze my progress towards my life goals and get advice.

I know, it's vague. So go ahead and close this issue. But at least I told you about this :-)

**Comment by tbabej** [comment]
This sounds to me more like a methodology than a feature, i.e. within the taskwarrior system this could be implemented as follows:
* "goals" and "milestones" are just different project hierarchies, i.e. `task add pro:goal.milestone.project`
* An external user-developed bash script could analyze the timewarior/taskwarrior data and give feedback on progress the goals

**Comment by kwoot** [comment]
Hi,

This is excellent! I did not know TW supports a project hierarchy like this. 

Yes, time permitting I will look into this more.

Thanks!

**Comment by tbabej** [comment]
Glad I could help!

**Comment by mrossinek** [comment]
I know you just closed the issue but I was in the process of writing this comment anyways, so I'll include it for possible future reference. I will also mainly include this because hierarchical projects are limited to linear hierarchies which may not always be sufficient.

I did some digging through the issues and found some possibly related ones. I haven't had the time yet to read through all of them but maybe @kwoot can provide some input in what regards they overlap with his idea:
- #533 (which means to link to #144)
- there is some discussion on milestones in #1282

**Comment by kwoot** [comment]
Apologies for closing too quickly. I was hesitant to open this issue because I know it is a pretty vague use-case.
The links to the related tickets are valuable. I will look into this.

**Comment by tbabej** [comment]
I think this is a good candidate for conversion into a discussion topic. Once we have a good idea what the action items are, we can spin off issues.
